crates/ruff_python_parser/src/string.rs

```diff
@@ -317,6 +320,92 @@ impl StringParser {
         })
     }
 
+    fn parse_tstring_middle(mut self) -> Result<ast::TStringLiteralElement, LexicalError> {
```

Comment:
Same here, What's the need for duplicating the entire parse logic? Can't we share most logic with the f-string handling?